OrbitGl/TimeGraph.cpp

```diff
@@ -390,42 +415,11 @@ void TimeGraph::GetWorldMinMax(float& a_Min, float& a_Max) const {
   a_Max = GetWorldFromTick(m_SessionMaxCounter);
 }
 
-//-----------------------------------------------------------------------------
-void TimeGraph::SelectLeft(const TextBox* a_TextBox) {
-  TextBox* textBox = const_cast<TextBox*>(a_TextBox);
-  Capture::GSelectedTextBox = textBox;
-  const Timer& timer = textBox->GetTimer();
-
-  if (IsVisible(timer)) {
-    return;
-  }
-
-  double currentTimeWindowUs = m_MaxTimeUs - m_MinTimeUs;
-  m_RefTimeUs = MicroSecondsFromTicks(m_SessionMinCounter, timer.m_Start);
-
-  double minTimeUs = m_RefTimeUs;
-  double maxTimeUs = m_RefTimeUs + currentTimeWindowUs;
-
-  SetMinMax(minTimeUs, maxTimeUs);
-}
-
-//-----------------------------------------------------------------------------
-void TimeGraph::SelectRight(const TextBox* a_TextBox) {
-  TextBox* textBox = const_cast<TextBox*>(a_TextBox);
-  Capture::GSelectedTextBox = textBox;
-  const Timer& timer = textBox->GetTimer();
-
-  if (IsVisible(timer)) {
-    return;
-  }
-  double currentTimeWindowUs = m_MaxTimeUs - m_MinTimeUs;
-  m_RefTimeUs = MicroSecondsFromTicks(m_SessionMinCounter, timer.m_End);
-
-  static double ratio = 1.0;
-  double minTimeUs = m_RefTimeUs - ratio * currentTimeWindowUs;
-  double maxTimeUs = m_RefTimeUs + (1 - ratio) * currentTimeWindowUs;
-
-  SetMinMax(minTimeUs, maxTimeUs);
+void TimeGraph::Select(const TextBox* a_TextBox) {
+  TextBox* text_box = const_cast<TextBox*>(a_TextBox);
```

Comment:
Is there a chance that we don't need the const_cast here? Probably we have to remove the `const` from several TextBox-pointers, but that feels more appropriate to me than having a const_cast here. (Yes, I know, that was here before, so not your fault, but maybe a good chance to clean that up. :-))

Reply:
ok, I will change it.

Reply:
For this, I have to erase several several const. All from ThreadTrack.h, all from GpuTrack.h... And it's only to erase this const_cast. Feels more appropiate, but too many changes.

On the other hand, we could change GSelectedTextBox to be "const TextBox *". What do you think?

Reply:
If that works for GSelectedTextBox then yes. That's an even better solution!

Reply:
Thanks for checking this out. I didn't want to impose more trouble. :see_no_evil: